ui_framework/src/global_styling/mixins/_namespace.scss

```diff
@@ -0,0 +1,99 @@
+// Namespace mixin
+// This forces component, child, modifier and state namespacing.
+// Check the styleguide for more detail.
+
+$childSeparator: '__';
+$modifierSeparator: '--';
+$stateSeparator: '.is';
+
+// Is this mixin itself nested within a mixin?
+@function isNested($selector) {
+    $selector: selectorToString($selector);
+    @if str-index($selector, $modifierSeparator) {
+        @return true;
+    } @else if str-index($selector, $stateSeparator){
+      @return true;
+    } @else if str-index($selector, $childSeparator){
+      @return true;
+    } @else {
+      @return false;
+    }
+}
+
+@function selectorToString($selector) {
+    $selector: inspect($selector); //cast to string
```

Comment:
Minor nit: can you fix the indentation so that it's just 2 spaces throughout this file?